### PR TITLE
add unit test and adjust code structure

### PR DESCRIPTION
**Pull request by HenryChen233**

### CODE REVIEW

File: request_handler.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+func check(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func readLines(path string) ([]string, error) {
+	file, err := os.Open(path)
+	check(err)
```
Comment (adavis10006) [inline request_handler.go]: Please don't wrap error handling.

File: request_handler.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
```
Comment (adavis10006) [inline request_handler.go]: Please remove the blank line. Error handling should take place immediately.

File: request_handler.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+func editHandler(w http.ResponseWriter, r *http.Request, title string) {
+	p, err := loadPage(title)
+	check(err)
+
+	renderTemplate(w, "edit", p)
+}
+
+func saveHandler(w http.ResponseWriter, r *http.Request, title string) {
+
```
Comment (adavis10006) [inline request_handler.go]: Please remove the blank line.

File: request_handler.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+// Copyright 2010 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+"bufio"
```
Comment (adavis10006) [inline request_handler.go]: Please use goimport to sort imports.

File: request_handler.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+"html/template"
+"io/ioutil"
+"log"
+"net/http"
+"os"
+"regexp"
+"strings"
+)
+
+type Page struct {
```
Comment (adavis10006) [inline request_handler.go]: For all export types, functions, and methods, please document it (godoc).

File: request_handler.go (hunk 1 above)
Comment (HenryChen233) [reply request_handler.go]: this has been corrected

File: request_handler.go (hunk 2 above)
Comment (HenryChen233) [reply request_handler.go]: This has been corrected

File: request_handler.go (hunk 5 above)
Comment (HenryChen233) [reply request_handler.go]: I would document this in google doc

**Pull request merged**